### PR TITLE
Display UPP identifiers in the output.  Little bit of optimising query

### PR DESCRIPTION
**Pull request by david-smith-ft**

### CODE REVIEW

File: concordances/cypher.go [hunk 1]
```diff
@@ -33,12 +33,9 @@ func (pcw CypherDriver) CheckConnectivity() error {
 }
 
 type neoReadStruct struct {
-	UUID                string        `json:"UUID"`
-	PrefUUID            string        `json:"prefUUID"`
-	NewModelTypes       []string      `json:"newModelTypes"`
-	NewModelIdentifiers neoIdentifier `json:"newModelIdentifiers"`
-	OldModelTypes       []string      `json:"oldModelTypes"`
-	OldModelIdentifiers neoIdentifier `json:"oldModelIdentifiers"`
+	UUID        string        `json:"UUID"`
```
Comment (peteclark-ft) [inline concordances/cypher.go]: Should this be `"uuid"` to be idiomatic json?

File: concordances/cypher.go [hunk 2]
```diff
@@ -33,12 +33,9 @@ func (pcw CypherDriver) CheckConnectivity() error {
...
+	Types       []string      `json:"types"`
+	Identifiers neoIdentifier `json:"identifiers"`
```
Comment (peteclark-ft) [inline concordances/cypher.go]: Logically `identifiers` would represent multiple ids? Not just one `neoIdentifier`? If `neoIdentifier` as a struct represents more than one id, that should be called `neoIdentifiers` instead

File: concordances/cypher.go [hunk 3]
```diff
@@ -52,15 +49,12 @@ func (pcw CypherDriver) ReadByConceptID(identifiers []string) (concordances Conc
...
-		MATCH (p:Concept)<-[:IDENTIFIES]-(ids:Identifier)
-		WHERE NOT ids:UPPIdentifier
-        OPTIONAL MATCH (p:Concept)-[:EQUIVALENT_TO]->(canonical:Concept)
-        OPTIONAL MATCH (leafNode:Concept)-[:EQUIVALENT_TO]->(canonical:Concept)
-        OPTIONAL MATCH (leafNode:Concept)<-[:IDENTIFIES]-(leafId:Identifier)
-        WHERE NOT leafId:UPPIdentifier
-	    RETURN canonical.prefUUID as prefUUID, p.uuid AS UUID, labels(p) AS oldModelTypes, labels(p) AS newModelTypes,
-	    {labels:labels(ids), value:ids.value} AS oldModelIdentifiers,
-	    {labels:labels(leafId), value:leafId.value} AS newModelIdentifiers
+		MATCH (p)-[:IDENTIFIES]-(ids:Identifier)
```
Comment (peteclark-ft) [inline concordances/cypher.go]: Is this no longer required to be a `Concept`?

File: concordances/cypher.go [hunk 4]
```diff
@@ -208,6 +195,10 @@ func mapAuthorityToAuthorityProperty(authority string) string {
 
 func mapAuthorityToIdentifierLabel(authority string) (label string) {
 	switch authority {
+	case UP_AUTHORITY:
```
Comment (peteclark-ft) [inline concordances/cypher.go]: This should probably be a map

File: concordances/cypher_test.go [hunk 5]
```diff
@@ -300,17 +346,23 @@ func TestNeoReadByAuthorityEmptyConcordancesWhenUnsupportedAuthority(t *testing.
 }
 
 func readConceptAndCompare(t *testing.T, expected Concordances, actual Concordances, testName string) {
-	sort.Slice(expected.Concordance, func(i, j int) bool {
+	sort.SliceStable(expected.Concordance, func(i, j int) bool {
 		return expected.Concordance[i].Concept.ID < expected.Concordance[j].Concept.ID
 	})
-	sort.Slice(expected.Concordance, func(i, j int) bool {
+	sort.SliceStable(expected.Concordance, func(i, j int) bool {
```
Comment (peteclark-ft) [inline concordances/cypher_test.go]: Not sure I like this duplication for expected and actual concordance sorting, but can't offer an alternative (that doesn't involve implementing `sort.Interface` on Concordances in 3 diff ways)

File: concordances/cypher.go (hunk 2 above)
Comment (david-smith-ft) [reply concordances/cypher.go]: Yeah, agreed.  Hangover from the previous model.

File: concordances/cypher.go (hunk 3 above)
Comment (david-smith-ft) [reply concordances/cypher.go]: The `Concept` constraint is defined in the first line of the query, so it's not needed again.

File: concordances/cypher.go (hunk 4 above)
Comment (david-smith-ft) [reply concordances/cypher.go]: There's no performance improvement by changing it to a map, and I think this is _slightly_ more readable than defining the map and then doing the lookup.

File: concordances/cypher_test.go (hunk 5 above)
Comment (david-smith-ft) [reply concordances/cypher_test.go]: I've extracted it so at least it's not repeated for both slices.

File: concordances/cypher.go (hunk 4 above)
Comment (peteclark-ft) [reply concordances/cypher.go]: Disagree - you're doing a key value lookup, which is exactly what maps exist for. Adding new mappings will require 2 lines for the constants, plus an addition to this switch statement, which is inefficient

File: concordances/cypher_test.go [hunk 6]
```diff
@@ -346,26 +346,23 @@ func TestNeoReadByAuthorityEmptyConcordancesWhenUnsupportedAuthority(t *testing.
...
 
-	sort.SliceStable(actual.Concordance, func(i, j int) bool {
-		return actual.Concordance[i].Concept.ID < actual.Concordance[j].Concept.ID
+	fullConcordanceSort(expected.Concordance)
+	fullConcordanceSort(actual.Concordance)
+
+	assert.True(t, reflect.DeepEqual(expected, actual), fmt.Sprintf("Actual aggregated concept differs from expected: Test: %v \n Expected: %v \n Actual: %v", testName, expected, actual))
+}
+
+func fullConcordanceSort(concordanceList []Concordance) {
```
Comment (peteclark-ft) [inline concordances/cypher_test.go]: This is nitpicky but `fullConcordanceSort` should be simpler (i.e. `sortConcordances`). Naming it `full` implies theres another alternative sort func that could be used, but there isn't one.

**Pull request merged**